app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +260,20 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        return serviceLocator.fxaRepo.accountState.subscribe {
+            when (it) {
+                FxaRepo.AccountState.AUTHENTICATED_WITH_PROFILE -> Unit //TODO
+                FxaRepo.AccountState.AUTHENTICATED_NO_PROFILE ->
+                    fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
```

Comment:
The state flow always transitions from one to the next:
- NOT_AUTHENTICATED
- AUTHENTICATED_NO_PROFILE (since the profile is fetched async after authentication so we don't have it at first)
- AUTHENTICATED_WITH_PROFILE

I'm concerned that we'll see a flicker when moving from AUTHENTICATED_NO_PROFILE to AUTHENTICATED_WITH_PROFILE. We should address this.

Maybe the simplest way would be to move the logic into the model: we could add a state for "fetching profile", which delays for 5 seconds or until the profile is fetched – whichever comes first. If it times out, it can move to a "failed to fetch profile" state (though we'd need to check when the profile fetch might be retryed by sync).

Reply:
Filed https://github.com/mozilla-mobile/firefox-tv/issues/2735

Reply:
Good idea to file a follow-up to keep the scope down here.